src/mongocxx/test/spec/unified_tests/runner.cpp

```diff
@@ -340,10 +340,22 @@ options::server_api create_server_api(document::view object) {
     return server_api_opts;
 }
 
-write_concern get_write_concern(const document::element& opts) {
-    if (!opts["writeConcern"])
-        return {};
+read_preference get_read_preference(const document::element& opts) {
+    read_preference rp;
+    auto read_pref = opts["readPreference"];
+    auto doc = read_pref.get_document().value;
+    auto mode = std::string(doc["mode"].get_string().value);
+    auto max_staleness_seconds = doc["maxStalenessSeconds"].get_int32().value;
+    rp.max_staleness(std::chrono::seconds(max_staleness_seconds));
```

Comment:
`maxStalenessSeconds` is an optional field. See the [Unified Test Format](https://github.com/mongodb/specifications/blob/31bbf60c4f3008d77fc5ca5287b165e5ea97a4a6/source/unified-test-format/unified-test-format.rst#commonoptions-readpreference) for reference.

```suggestion
    if (doc["maxStalenessSeconds"]) {
        auto max_staleness_seconds = doc["maxStalenessSeconds"].get_int32().value;
        rp.max_staleness(std::chrono::seconds(max_staleness_seconds));
    }
```